src/Handler/CloudWatch.php

```diff
@@ -7,139 +7,70 @@
-
-    /**
-     * @var CloudWatchLogsClient
-     */
-    private $client;
-
-    /**
-     * @var string
-     */
-    private $group;
-
-    /**
-     * @var string
-     */
-    private $stream;
-
-    /**
-     * @var integer
-     */
-    private $retention;
-
-    /**
-     * @var bool
-     */
-    private $initialized = false;
-
-    /**
-     * @var string
-     */
-    private $sequenceToken;
-
-    /**
-     * @var int
-     */
-    private $batchSize;
-
-    /**
-     * @var array
-     */
-    private $buffer = [];
-
-    /**
-     * @var array
-     */
-    private $tags = [];
-
-    /**
-     * @var bool
      */
-    private $createGroup;
+    public const TIMESPAN_LIMIT = 86400000;
+
+    private CloudWatchLogsClient $client;
+    private string $group;
+    private string $stream;
+    private int $retention;
+    private bool $initialized = false;
+    private string | null $sequenceToken;
+    private int $batchSize;
+    /** @var LogRecord[] $buffer */
+    private array $buffer = [];
+    private array $tags = [];
+    private bool $createGroup;
 
     /**
      * Data amount limit (http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html)
-     *
-     * @var int
-     */
-    private $dataAmountLimit = 1048576;
-
-    /**
-     * @var int
-     */
-    private $currentDataAmount = 0;
-
-    /**
-     * @var int
-     */
-    private $remainingRequests = self::RPS_LIMIT;
-
-    /**
-     * @var \DateTime
-     */
-    private $savedTime;
-
-    /**
-     * @var int|null
      */
-    private $earliestTimestamp = null;
+    private int $dataAmountLimit = 1048576;
+    private int $currentDataAmount = 0;
+    private int $remainingRequests = self::RPS_LIMIT;
+    private \DateTime $savedTime;
+    private int | null $earliestTimestamp = null;
 
     /**
      * CloudWatchLogs constructor.
-     * @param CloudWatchLogsClient $client
      *
      *  Log group names must be unique within a region for an AWS account.
      *  Log group names can be between 1 and 512 characters long.
      *  Log group names consist of the following characters: a-z, A-Z, 0-9, '_' (underscore), '-' (hyphen),
      * '/' (forward slash), and '.' (period).
-     * @param string $group
      *
      *  Log stream names must be unique within the log group.
      *  Log stream names can be between 1 and 512 characters long.
      *  The ':' (colon) and '*' (asterisk) characters are not allowed.
-     * @param string $stream
-     *
-     * @param int $retention
-     * @param int $batchSize
-     * @param array $tags
-     * @param int $level
-     * @param bool $bubble
-     * @param bool $createGroup
-     *
      * @throws \Exception
      */
     public function __construct(
         CloudWatchLogsClient $client,
-        $group,
-        $stream,
-        $retention = 14,
-        $batchSize = 10000,
+        string $group,
+        string $stream,
+        int $retention = 14,
```

Comment:
I thought retention could be null for infinite. So that'd be int?

Reply:
@noahred16 Apologies but many months ago I forked this repository and applied the above PR, with additional fixes and improvement. It's available at [phpnexus/cwh](https://github.com/phpnexus/cwh).

I intend to maintain my fork as I use CloudWatch Logs with PHP 8.1 at my workplace, but may I ask if you plan to take over maintenance of this original repository?

Reply:
Hi @markinjapan. No, I just started using CloudWatch with my project and found this repo useful. I commented here but I did not realize that this repo was inactive. I'll take a look at yours. Thanks for sharing.